lib/ansible/modules/system/osx_defaults.py

```diff
@@ -15,413 +16,975 @@
 DOCUMENTATION = '''
 ---
 module: osx_defaults
-author: Franck Nijhof (@frenck)
-short_description: osx_defaults allows users to read, write, and delete Mac OS X user defaults from Ansible
+author: "Franck Nijhof (@frenck), Etienne Desautels (@etienned)"
+short_description:
+    Allows users to read, write, and delete Mac OS X preferences from Ansible.
 description:
-  - osx_defaults allows users to read, write, and delete Mac OS X user defaults from Ansible scripts.
-    Mac OS X applications and other programs use the defaults system to record user preferences and other
-    information that must be maintained when the applications aren't running (such as default font for new
-    documents, or the position of an Info panel).
+  - osx_defaults allows users to read, write, and delete Mac OS X preferences
+    from Ansible scripts.
+    Mac OS X applications and other programs use the defaults system to record
+    user preferences and other information that must be maintained when the
+    applications aren't running (such as default font for new documents, or
+    the position of an Info panel).
 version_added: "2.0"
+requirements:
+  - Python 2.6+.
+  - OS X installed Foundation and CoreFoundation python modules (or from PyObjC).
 options:
   domain:
     description:
-      - The domain is a domain name of the form com.companyname.appname.
-    required: false
+      - The I(domain) is a domain name of the form com.companyname.appname or
+        I(NSGlobalDomain) for the Global Domain.
     default: NSGlobalDomain
   host:
```

Comment:
I suggest adding `choices: ["anyHost", "currentHost"]` with  `default: anyHost`

Reply:
I don't think I should do this for two reasons:

1. I want the `osx_defaults` module to be backward compatible, so peoples already using it do not have to rewrite their tasks to use this module that have more features.
2. It's possible to specify a specific hostname as host, so using `choices` is not an option.